samples/subsys/task_wdt/boards/nucleo_f091rc.overlay

```diff
@@ -0,0 +1,14 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2022 STMicroelectronics
+ */
+
+/*
+ * Due to the LSI RC oscillator characteristics (defined by the datasheet),
+ * the typical value set in the soc or board dts might be too low
+ * for the IWDG timeout : define the max value for this usecase.
+ */
+&clk_lsi {
+	clock-frequency = <DT_FREQ_K(50)>;
+};
```

Comment:
If this overlay is required always if the task watchdog is used, this is a very hacky approach and not a valid solution. The IWDG and task watchdog must work with default IWDG settings coming from SOC .dts. Either the task watchdog has to be changed to accept higher tolerances or this setting must be configured for *all* relevant STM32 MCUs in the SOC .dts files, not in an overlay. And it should probably called `max-clock-frequency` in that case, not `clock-frequency`, which is the nominal/average if I understand correctly.

Reply:
Only the conversion of IWDG timeout in ticks, in iwdg_stm32_convert_timeout function, requires the LSI MAXimum value as given by the datasheet (typical LSI freq might be too low in for several devices). 

```
	/* Convert timeout to LSI clock ticks. */
	uint32_t ticks = (uint64_t)timeout * STM32_LSI_FREQ / USEC_PER_SEC;
```

In the <clk-lsi> node of the soc DTS, we could add a <MAX-clock-frequency> property so that IWDG driver use that value in the calculation as above --> impacting the stm32 IWDG driver only
Or we always give the MAX value to the <clock-frequency> of the <clk-lsi> node --> impacting all the users of lsi freq.

Reply:
Introducing a compatible: "st,stm32-lsi-clock" for all the stm32 devices with a `<max-clock-frequency>` 

then will the wdt_idwg_stm32.c use the STM32_MAX_LSI_FREQ
```
/* Convert timeout to LSI clock ticks. */
uint32_t ticks = (uint64_t)timeout * STM32_LSI_FREQ / USEC_PER_SEC;
```